test/common/upstream/od_cds_api_impl_test.cc

```diff
@@ -0,0 +1,137 @@
+#include "envoy/config/core/v3/config_source.pb.h"
+#include "envoy/config/subscription.h"
+
+#include "common/stats/isolated_store_impl.h"
+#include "common/upstream/od_cds_api_impl.h"
+
+#include "test/mocks/protobuf/mocks.h"
+#include "test/mocks/upstream/cluster_manager.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Upstream {
+namespace {
+
+using ::testing::ElementsAre;
+using ::testing::InSequence;
+using ::testing::UnorderedElementsAre;
+
+class OdCdsApiImplTest : public testing::Test {
+public:
+  void SetUp() override {
+    envoy::config::core::v3::ConfigSource odcds_config;
+    OptRef<xds::core::v3::ResourceLocator> null_locator;
+    odcds_ = OdCdsApiImpl::create(odcds_config, null_locator, cm_, store_, validation_visitor_);
+    odcds_callbacks_ = cm_.subscription_factory_.callbacks_;
+  }
+
+  NiceMock<MockClusterManager> cm_;
+  Stats::IsolatedStoreImpl store_;
+  OdCdsApiSharedPtr odcds_;
+  Config::SubscriptionCallbacks* odcds_callbacks_ = nullptr;
+  NiceMock<ProtobufMessage::MockValidationVisitor> validation_visitor_;
+};
+
+TEST_F(OdCdsApiImplTest, FirstUpdateStarts) {
```

Comment:
Could you add comments explaining what's being verified?

Reply:
Sure.